autoPyTorch/data/tabular_feature_validator.py

```diff
@@ -247,10 +256,12 @@ def transform(
         # We need to convert the column in test data to
         # object otherwise the test column is interpreted as float
         if len(self.categorical_columns) > 0:
-            categorical_columns = self.column_transformer.transformers_[0][-1]
-            for column in categorical_columns:
-                if X[column].isna().all():
-                    X[column] = X[column].astype('object')
+            assert hasattr(self, 'enc_columns') and self.enc_columns is not None  # MyPy
```

Comment:
we have changed this to `transformed_columns`, I think we should wait for PR #306 to be merged before merging this one as well.